pkg/test/framework/components/istio/operator.go

```diff
@@ -231,6 +247,34 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	return i, nil
 }
 
+func patchIstiodCustomHost(c *operatorComponent, cfg Config, cluster kube.Cluster) error {
+	var remoteIstiodAddress net.TCPAddr
+	if err := retry.UntilSuccess(func() error {
+		var err error
+		remoteIstiodAddress, err = GetRemoteDiscoveryAddress(cfg.SystemNamespace, cluster, false)
+		return err
+	}, retry.Timeout(1*time.Minute)); err != nil {
+		return fmt.Errorf("failed getting the istiod address for cluster %d: %v", cluster.Index(), err)
+	}
+
+	if err := cluster.PatchDeployment(cfg.ConfigNamespace, "istiod", fmt.Sprintf(`
```

Comment:
We need to be testing what users are actually doing. I hope that we are not recommending them to use `kubectl patch`

Reply:
Meaning it should be set in the `--set` options

Reply:
You can use --set for the ISTIOD_CUSTOM_HOST env var - but that only works if you know the IP of ingress gateway prior to deploy.  There is an item for making this item configurable: https://github.com/istio/api/pull/1414

Reply:
yea unfortunatelly  this is what we do before a proper API is in place, unless the gw address is known ahead of time.